### PR TITLE
[AUCT-952] Bidder sees copy in "about to register" and "registration pending" modals with information about identity verification

### PR DESCRIPTION
**Pull request by yuki24**
addresses https://artsyproduct.atlassian.net/browse/AUCT-952

This PR is coupled to https://github.com/artsy/force/pull/5392. Please see the other PR for more details.

### CODE REVIEW

File: src/Components/Auction/PostRegistrationModal.tsx [hunk 1]
```diff
@@ -35,6 +36,26 @@ const RegistrationPending: ModalContent = ({ onClick }) => {
...
+
+const RegistrationPendingUnverified: ModalContent = ({ onClick }) => {
+  return (
+    <>
+      <RegistrationPendingHeader />
+      <Serif my={3} size="3t">
+        This auction requires Artsy to verify your identity before bidding.
+        <br />
+        <br />
+        For details about identity verification, please see the FAQ or contact
```
Comment (bhoggard) [inline src/Components/Auction/PostRegistrationModal.tsx:48]: Is `FAQ` supposed to be a link here?

Comment (yuki24) [reply src/Components/Auction/PostRegistrationModal.tsx:48]: It should be a link, added a todo to https://github.com/artsy/force/pull/5392.

File: src/Components/Auction/PostRegistrationModal.tsx [hunk 2]
```diff
@@ -35,6 +36,32 @@ const RegistrationPending: ModalContent = ({ onClick }) => {
...
+    <>
+      <RegistrationPendingHeader />
+      <Serif my={3} size="3t">
+        This auction requires Artsy to verify your identity before bidding.
+        <br />
+        <br />
+        For details about identity verification, please see the
+        <a
+          target="_blank"
+          href="https://www.artsy.net/identity-verification-faq"
```
Comment (dleve123) [inline src/Components/Auction/PostRegistrationModal.tsx:51]: can we scope this to the environment that we're in?  I believe we can use `getENV('APP_URL')`, similar to what we do here:

https://github.com/artsy/reaction/blob/master/src/Apps/Artwork/Components/ArtworkSidebar/ArtworkSidebarBidAction.tsx#L103

File: src/Components/Auction/__tests__/AuctionRegistrationModal.test.tsx [hunk 3]
```diff
@@ -11,31 +11,82 @@ const closeMock = jest.fn()
...
+      },
+    })
+
+    expect(wrapper.find(Modal).text()).toMatch("Register for IDV Sale")
+    expect(wrapper.find(Modal).text()).toMatch(
+      "Welcome back. To complete your registration, please confirm that you agree to the Conditions of Sale."
+    )
+  })
+
+  it("shows an IDV registration message if the sale requires IDV and the user is verified", async () => {
```
Comment (dleve123) [inline src/Components/Auction/__tests__/AuctionRegistrationModal.test.tsx:62]:
```suggestion
  it("shows an IDV registration message if the sale requires IDV and the user is not verified", async () => {
```

File: src/Utils/identityVerificationRequirements.ts [hunk 4]
```diff
@@ -25,8 +25,8 @@ export const bidderNeedsIdentityVerification = ({
   bidder,
 }: {
   sale: IdentityVerificationRequireable
-  user: IdentityVerifiable | null
-  bidder: Bidder | null
+  user?: IdentityVerifiable
```
Comment (dleve123) [inline src/Utils/identityVerificationRequirements.ts:28]: What motivates this diff? This makes sense for bidder, as sometimes there is no bidder object within scope, but there's always a `me` for us to consider (even if that me is `null`). I'm probably just not understanding some nuance of TS here, but thanks for clarifying!

Comment (yuki24) [reply src/Utils/identityVerificationRequirements.ts:28]: Having both `value: Type | null` and `value?: Type` next to each other makes understanding the signature quite tricky. Also the `?` symbol aligns more with TS's new optional chaining `value?.attr` both semantically and visually.

File: src/Components/Auction/PostRegistrationModal.tsx (hunk 2 above)
Comment (yuki24) [inline src/Components/Auction/PostRegistrationModal.tsx:51]:
```suggestion
          href="/identity-verification-faq"
```

Comment (yuki24) [reply src/Components/Auction/PostRegistrationModal.tsx]: just updated to use a relative path.

File: src/Utils/identityVerificationRequirements.ts [hunk 5]
```diff
@@ -25,8 +25,8 @@ export const bidderNeedsIdentityVerification = ({
...
+  bidder?: Bidder
 }) => {
   return (
     !bidder?.qualified_for_bidding &&
```
Comment (erikdstock) [inline src/Utils/identityVerificationRequirements.ts:32]: I realize this is not part of the diff but why is it necessary to check if the bidder is qualified for bidding? Wouldn't that be encapsulated by the checks below?

**Comment by artsyit** [comment]
:rocket: PR was released in `v26.17.3` :rocket:

File: src/Utils/identityVerificationRequirements.ts (hunk 5 above)
Comment (dleve123) [reply src/Utils/identityVerificationRequirements.ts:32]: @erikdstock Check out [this slack thread](https://artsy.slack.com/archives/CA3D02SDN/p1586546440286600) and [this bug](https://artsyproduct.atlassian.net/browse/AUCT-985) – both should help clarify why this is needed.

**Pull request merged**